src/Components/Shared/src/WebRootComponentParameters.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal readonly struct WebRootComponentParameters(
+    ParameterView parameterView,
+    IReadOnlyList<ComponentParameter> parameterDefinitions,
+    IReadOnlyList<object> serializedParameterValues)
```

Comment:
Should this be string?

Reply:
` IReadOnlyList<JsonElement>`

Reply:
This matches the existing pattern we follow here:
https://github.com/dotnet/aspnetcore/blob/86770ce5945ae3b5651030d97af40d7c701c50b9/src/Shared/Components/ServerComponent.cs#L36
But we could probably convert it to an `IReadOnlyList<JsonElement>` for this case

Reply:
Hm, we actually probably can't do `IReadOnlyList<JsonElement>` without constructing an entirely new list. Do you think it's fine if we leave this as-is?